indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -210,12 +215,16 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
    * @param preferredVersion use this version string if one has not yet been assigned
    *
    * @return lock version if lock was acquired, absent otherwise
+   * @throws IllegalStateException if the task is not a valid active task
    */
   private Optional<TaskLock> tryLock(final Task task, final Interval interval, final Optional<String> preferredVersion)
   {
     giant.lock();
 
     try {
+      if(!tasks.contains(task)){
```

Comment:
This has me a bit worried because then every task needs to implement equals and hash properly, which I'm not sure was enforced prior to this. Is there a reason it cannot be a set of the string TaskID?